pkg/auth/dependency/interaction/provider_intent.go

```diff
@@ -0,0 +1,23 @@
+package interaction
+
+import (
+	"fmt"
+
+	"github.com/skygeario/skygear-server/pkg/auth/dependency/auth"
+)
+
+func (p *Provider) NewInteraction(intent Intent, clientID string, session auth.AuthSession) (*Interaction, error) {
```

Comment:
IMO it is simpler to have separate functions to create Interaction with a concrete Intent.

```golang
func InteractionWithIntentLogin(intent *IntentLogin, clientID string) *Interaction {
  return &Interaction{
    Intent: intent,
    ClientID: clientID,
  }
}
```